app/controllers/posts_controller.rb

```diff
@@ -7,4 +10,33 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :text)
```

Comment:
- [ ] please make sure the post is created on behalf of the current_user which is not the case at the moment.

Reply:
- Well resolved